gridcapa-core-cc-post-processing-app/src/main/java/com/farao_community/farao/core_cc_post_processing/app/services/F305XmlGenerator.java

```diff
@@ -124,9 +124,11 @@ private static void generateRaoResponsePayLoad(Set<TaskDto> taskDtos, ResponseMe
                 boolean includeResponseItem = true;
 
                 if (taskDto.getStatus().equals(TaskStatus.ERROR)) {
-                    if (!metadataMap.containsKey(taskDto.getId()) || StringUtils.equals(metadataMap.get(taskDto.getId()).getErrorMessage(), "Missing raoRequest")) {
+                    if (metadataMap.containsKey(taskDto.getId()) && StringUtils.equals(metadataMap.get(taskDto.getId()).getErrorMessage(), "Missing raoRequest")) {
                         // Do not generate a responseItem : raoRequest was not defined for this timestamp
                         includeResponseItem = false;
+                    } else if (!metadataMap.containsKey(taskDto.getId())) {
+                        fillFailedHours(responseItem, "500-InternalException", "No output available");
```

Comment:
As a way of reducing complexity you could rewrite if(A && B) else if (!A) into if(!A) else if (B)
```suggestion
                    if (!metadataMap.containsKey(taskDto.getId())) {
                        fillFailedHours(responseItem, "500-InternalException", "No output available");
                    } else if (StringUtils.equals(metadataMap.get(taskDto.getId()).getErrorMessage(), "Missing raoRequest")) {
                        // Do not generate a responseItem : raoRequest was not defined for this timestamp
                        includeResponseItem = false;
                    }
```